subworkflows/kraken_pipeline.nf

```diff
@@ -9,42 +9,49 @@ workflow kraken_pipeline {
     main:
 
         // Check source param is valid
-        sources = params.database_sets
-        source_name = params.database_set
-        source_data = sources.get(source_name, false)
-        if (!sources.containsKey(source_name) || !source_data) {
-            keys = sources.keySet()
-            throw new Exception("Source $params.source is invalid, must be one of $keys")
-        }
+        // sources = params.database_sets
+        // source_name = params.database_set
+        // source_data = sources.get(source_name, false)
+        // if (!sources.containsKey(source_name) || !source_data) {
+        //     keys = sources.keySet()
+        //     throw new Exception("Source $params.source is invalid, must be one of $keys")
+        // }
 
         // Grab taxonomy files
-        taxonomy = file(sources[source_name]["taxonomy"], type: "file")
-        if (params.taxonomy) {
-            log.info("Checking custom taxonomy mapping exists")
-            taxonomy = file(params.taxonomy, type: "dir", checkIfExists:true)
-        }
+        // taxonomy = file(sources[source_name]["taxonomy"], type: "file")
+        // if (params.taxonomy) {
+        //     log.info("Checking custom taxonomy mapping exists")
+            
+        // }
+
+        taxonomy = file(params.taxonomy_dir, type: "dir", checkIfExists:true)
 
-        if (database) {
-            source_database = database
-        } else {
-            source_database = source_data.get("database", false)
-            if (!source_database) {
-                throw new Exception(
-                    "Error: Source $source_name does not include a database for "
-                    + "use with kraken, please choose another source, "
-                    + "provide a custom database or disable kraken.")
-            }
-        database = file(source_database, type: "file")
+        // if (database) {
+        //     source_database = database
+        // } else {
+        //     source_database = source_data.get("database", false)
+        //     if (!source_database) {
+        //         throw new Exception(
+        //             "Error: Source $source_name does not include a database for "
+        //             + "use with kraken, please choose another source, "
+        //             + "provide a custom database or disable kraken.")
+        //     }
+        database = file(params.db, type: "dir", checkIfExists:true)
 
         // start_server(database, taxonomy)
         run_kraken_and_bracken(unique_id, fastq, database, taxonomy)
```

Comment:
One thing we need to add is some sort of handling for if kraken produces a very low number of hits causing bracken to fail out and crash the pipe